netbox_onboarding/onboard.py

```diff
@@ -430,18 +429,15 @@ def ensure_device_type(
             self.manufacturer = Manufacturer.objects.create(name=self.netdev.vendor, slug=slugify(self.netdev.vendor))
             self.manufacturer.save()
 
-        # Now see if the device type (slug) already exists,
+        # Now see if the device type already exists by searching for its slug, model or part number
         #  if so check to make sure that it is not assigned as a different manufacturer
-        # if it doesn't exist, create it if the flag 'create_device_type_if_missing' is defined
-
-        slug = self.netdev.model
-        if re.search(r"[^a-zA-Z0-9\-_]+", slug):
-            logging.warning("device model is not sluggable: %s", slug)
-            self.netdev.model = slug.replace(" ", "-")
-            logging.warning("device model is now: %s", self.netdev.model)
-
+        #  if it doesn't exist, create it if the flag 'create_device_type_if_missing' is defined
         try:
-            self.device_type = DeviceType.objects.get(slug=slugify(self.netdev.model))
+            self.device_type = DeviceType.objects.get(
+                Q(slug__iexact=slugify(self.netdev.model))
+                | Q(model__iexact=self.netdev.model)
+                | Q(part_number__iexact=self.netdev.model)
```

Comment:
Do you need to handle the case where this could match on multiple devicetypes now? (I.e, one devicetype's model name is the same as another devicetype's slug, so we now have two "matching" devicetypes?) I don't know how realistic that is but it seems like something that could be potentially encountered.

Reply:
Good point, I added a new section to catch the exception if multiple objects are returned, thanks